beam/beam/scan/user_login.py

```diff
@@ -0,0 +1,39 @@
+import frappe
+from beam.beam.scan import get_barcode_context
+from frappe.core.doctype.user.user import get_restricted_ip_list
+from erpnext import get_default_company
+
+
+@frappe.whitelist(allow_guest=True)
+def scan_login(barcode):
+	client_ip = (
+		frappe.local.request.headers.get("X-Forwarded-For") or frappe.local.request.remote_addr
+	)
+
+	user = get_barcode_context(barcode)
+	if not user:
+		frappe.throw("Wrong barcode", title="Login Error")
+
+	if user["doc"].doctype != "User":
+		frappe.throw("Wrong barcode", title="Login Error")
+
+	company = get_default_company()
```

Comment:
First, check if the user has a related employee and use that employee's company. If not, fall back to the default company.